protocol_rfcs/catalog-owned.md

```diff
@@ -0,0 +1,489 @@
+./_delta_log/_staged_commits/00000000000000000002.3ae45b72-24e1-865a-a211-34987ae02f2a.json
+```
+
+<ins>NOTE: The (proposed) version number of a staged commit is authoritative - file
+`00000000000000000100.<uuid>.json` always corresponds to a commit attempt for version 100. Besides
+simplifying implementations, it also acknowledges the fact that commit files cannot safely be reused
+for multiple commit attempts. For example, resolving conflicts in a table with [row
+tracking](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#row-tracking) enabled requires
+rewriting all file actions to update their `baseRowId` field.</ins>
+
+<ins>The [catalog](#terminology-catalogs) is the source of truth about which staged commit files in
+the `_delta_log/_staged_commits` directory correspond to ratified versions, and Delta clients should
+not attempt to directly interpret the contents of that directory. Refer to
+[catalog-owned tables](#catalog-owned-tables) for more details.</ins>
+
+~~Delta files use new-line delimited JSON format, where every action is stored as a single line JSON
+document. A delta file, `n.json`, contains an atomic set of [_actions_](#actions) that should be
+applied to the previous table state, `n-1.json`, in order to the construct `n`th snapshot of the
+table. An action changes one aspect of the table's state, for example, adding or removing a file.~~
+
+<ins>Delta files use newline-delimited JSON format, where every action is stored as a single-line
+JSON document. A Delta file, corresponding to version `v`, contains an atomic set of
+[_actions_](#Actions) that should be applied to the previous table state corresponding to version
+`v-1`, in order to construct the `v`th snapshot of the table. An action changes one aspect of the
+table's state, for example, adding or removing a file.</ins>
+
+### Commit Provenance Information
+
+> ***Change to [existing section](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#commit-provenance-information)***
+
+<ins>When the `catalogOwned` table feature is enabled, the `commitInfo` action must have a field
+`txnId` that stores a unique transaction identifier string.</ins>
+
+### Metadata Cleanup
+
+> ***Change to [existing section](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#metadata-cleanup)***
+
+2. Identify the newest checkpoint that is not newer than the `cutOffCommit`. A checkpoint at the
+   `cutOffCommit` is ideal, but an older one will do. Lets call it `cutOffCheckpoint`. We need to
+   preserve the `cutOffCheckpoint` and all <ins>published</ins> commits after it, because we need
+   them to enable time travel for commits between `cutOffCheckpoint` and the next available
+   checkpoint.
+    - <ins>If no `cutOffCheckpoint` can be found, do not proceed with metadata cleanup as there is
+      nothing to cleanup.</ins>
+3. Delete all [delta log entries](#delta-log-entries) and [checkpoint files](#checkpoints) before
```

Comment:
Existing: should we have the CRC files listed here as well for deletion?

Reply:
Yes!